modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/session.js

```diff
@@ -236,113 +236,61 @@ AUI.add(
 				},
 
 				_startTimer() {
-					var instance = this;
-
-					var sessionLength = instance.get('sessionLength');
+					const instance = this;
 
-					var sessionTimeoutOffset = instance.get(
+					const sessionLength = instance.get('sessionLength');
+					const sessionTimeoutOffset = instance.get(
 						'sessionTimeoutOffset'
 					);
-
-					var warningTime = instance.get('warningTime');
-
-					var registered = instance._registered;
-
-					var interval = 1000;
+					const warningTime = instance.get('warningTime');
 
 					instance._intervalId = A.setInterval(() => {
-						var sessionState = instance.get('sessionState');
+						const sessionState = instance.get('sessionState');
+						const timestamp = instance.get('timestamp');
 
-						var timeOffset;
+						// LPS-82336 Maintain session state in multiple tabs
 
-						var timestamp = instance.get('timestamp');
+						if (instance._initTimestamp !== timestamp) {
+							instance.set('timestamp', timestamp);
 
-						var elapsed = sessionLength;
-
-						if (Lang.toInt(timestamp)) {
-							timeOffset =
-								Math.floor((Date.now() - timestamp) / 1000) *
-								1000;
-
-							elapsed = timeOffset;
-
-							if (instance._initTimestamp !== timestamp) {
-								instance.set('timestamp', timestamp);
-
-								if (sessionState !== 'active') {
-									instance.set(
-										'sessionState',
-										'active',
-										SRC_EVENT_OBJ
-									);
-								}
+							if (sessionState !== 'active') {
+								instance.set(
+									'sessionState',
+									'active',
+									SRC_EVENT_OBJ
+								);
 							}
 						}
-						else {
-							timestamp = 'expired';
-						}
 
-						var extend = instance.get('autoExtend');
+						const elapsed =
+							Math.floor((Date.now() - timestamp) / 1000) * 1000;
 
-						var expirationMoment = false;
-						var warningMoment = false;
+						const autoExtend = instance.get('autoExtend');
 
-						var hasExpired = elapsed >= sessionLength;
-						var hasExpiredTimeoutOffset =
+						const hasExpired = elapsed >= sessionLength;
+						const hasExpiredTimeoutOffset =
 							elapsed >= sessionLength - sessionTimeoutOffset;
-						var hasWarned = elapsed >= warningTime;
-
-						if (hasExpiredTimeoutOffset || hasWarned) {
-							if (timestamp === 'expired') {
-								expirationMoment = true;
-								extend = false;
-								hasExpired = true;
-								hasExpiredTimeoutOffset = true;
-							}
+						const hasWarned = elapsed >= warningTime;
 
-							if (
-								hasExpiredTimeoutOffset &&
-								sessionState !== 'expired'
-							) {
-								if (extend && !hasExpired) {
-									expirationMoment = false;
-									hasExpired = false;
-									hasExpiredTimeoutOffset = false;
-									hasWarned = false;
-									warningMoment = false;
-
-									instance.extend();
-								}
-								else {
-									instance.expire();
+						if (autoExtend && hasExpiredTimeoutOffset) {
+							instance.extend();
+						}
```

Comment:
Hi @markocikos 

I think your changes will cause https://issues.liferay.com/browse/LPS-137264 to be reproduced again.

You should change this code with:
```
if (autoExtend && hasExpiredTimeoutOffset) {
	if (hasExpired) {
		instance.expire();
	}
	else {
		instance.extend();
	}
}
```

As we should only send the extend session request during the offset window:

- elapsed >= sessionLength - sessionTimeoutOffset
- elapsed < sessionLength

Reply:
@jorgediaz-lr Thank you very much for checking in on the changes!

You are right, this would have a been a regression. Personally, I wouldn't mind if session was extended even if hops over `sessionTimeout`. But, I'll respect LPS-137264 as definitive for expected behavior.

I added a commit with code that achieves what you suggested above, only SFd to be more concise. I retested all the use cases from opening post, and the one in LPS-137264. Btw, nice trick with `pkill -STOP -f chrome`! :sparkles: